classes/class-wc-connect-api-client.php

```diff
@@ -372,11 +372,14 @@ protected function request( $method, $path, $body = array() ) {
 			// If the received response is not JSON, return the raw response
 			$content_type = wp_remote_retrieve_header( $response, 'content-type' );
 			if ( false === strpos( $content_type, 'application/json' ) ) {
-				if ( 200 != $response_code ) {
-					return new WP_Error(
-						'wcc_server_error',
-						sprintf( 'Error: The WooCommerce Services server returned HTTP code: %d', $response_code )
-					);
+				if ( ! $response_code ) {
+					return new WP_Error( 'wcc_server_error', 'The WooCommerce Services API could not be reached.' );
+				} elseif ( 500 == $response_code ) {
+					return new WP_Error( 'wcc_server_error', 'The WooCommerce Services API encountered an error. Please try again.' );
+				} elseif ( 500 < $response_code ) {
+					return new WP_Error( 'wcc_server_error', 'The WooCommerce Services API is unavailable. Please try again in just a moment.' );
+				} elseif ( 400 <= $response_code ) {
```

Comment:
I don't know how common/possible it would be in non-development environments, but I see 404s a lot on the order details page when switching between my local WCS server and staging/production.

This change removes the meaningful "not found" I saw before.

Reply:
:+1: Will bring back that string for `4xx` responses at least.